src/fi/bittiraha/walletd/WalletRPC.java

```diff
@@ -148,6 +149,18 @@ private String getnewaddress() {
     return address;
   }
 
+  private BigDecimal getreceivedbyaddress(String address) {
+    Coin retval = Coin.ZERO;
+	List<TransactionOutput> unspent = kit.wallet().calculateAllSpendCandidates(false, false);
```

Comment:
calculateAllSpendCandidates returns a list of unspent outputs. You'll need to scan all transactions to implement the equivalent of bitcoind does.